### ISSUE TITLE
[Question] How do we pass authentication data to a web hook handler?

### ISSUE THREAD

**Issue by elkdanger** [aspnet/WebHooks#21]
For example, in the Instagram web hook handler example. On the back of handling a web hook request, if I wanted to then connect to Instagram's API to retrieve more information about the image I would need an access token. I have the token currently stored in the session, but what's the pattern for getting such a thing through to the handler from my application?

**Comment by HenrikFrystykNielsen** [comment]
Just to answer your cookie question first: you can get your cookies from the HttpRequestMessage in the context passed to the handler as described here [1].

To clarify a bit on where tokens are needed, Instagram has four different ways you can subscribe: based on users, tags, location-tagging, and geo-location. For geo-location subscriptions you don't need a token to get additional information about the images (this is what you see in the sample where we use the InstagramWebHookClient to get additional information about the images). 

However, for the other kinds of subscriptions you do need a token representing a particular user. Once you have a token, you can submit your own request to Instagram for additional information about the image. InstagramWebHookClient only supports getting additional information for geo-location subscriptions.

Henrik

[1] http://www.asp.net/web-api/overview/advanced/http-cookies

**Comment by elkdanger** [comment]
Thanks for the response. Yes it was when I tried to break out of the Geo example that I realised I needed to do extra work to get data from Instagram regarding images that had come from a tag search. So far, I'm no further forward with passing the access token through, as even after following the article you linked to, I still can't retrieve the session cookie. It just doesn't appear in any cookie collection that I can get access to. The only one that consistently appears in the collection is "ARRAffinity".

I also realise that this issue is not so much to do with WebHooks as with WebApi. I don't suppose there are any examples around which demonstrate what I'm trying to achieve?

Essentially, this is my workflow:

* Redirecting to Instagram for OAuth from MVC
* Returning from Instagram, retrieving the auth token then `Response.Cookies.Add(new HttpCookie("AuthToken", response.AccessToken))`
* Created a class inheriting from `DelegatingHandler` to try and read the cookie in, and registered the handler with WebApi
* Made a simple POST request to the `Subscribe` endpoint (as per Instagram example) and inspected the cookies collection from `request.Headers.GetCookies()`

I feel like I'm missing a piece here - should I be able to see the cookies coming through when I debug the endpoint in the last step? Should they then filter through to the web hook handler, when that request is made by Instagram?
